relay/src/setup.rs

```diff
@@ -185,18 +164,36 @@ pub fn init_logging(config: &Config) {
     };
 
     let log = Box::new(log_builder.build());
-    let global_filter = log.filter();
-
-    sentry::integrations::log::init(
-        Some(log),
-        sentry::integrations::log::LoggerOptions {
-            global_filter: Some(global_filter),
-            attach_stacktraces: config.enable_backtraces(),
-            ..Default::default()
-        },
-    );
 
-    sentry::integrations::panic::register_panic_handler();
+    let log_integration = sentry::integrations::log::LogIntegration {
+        global_filter: Some(log.filter()),
+        attach_stacktraces: true,
+        dest_log: Some(log),
+        ..Default::default()
+    };
+
+    let guard = sentry::init(sentry::ClientOptions {
```

Comment:
Moved `sentry::init` down so that we can pass the log integration in. This comes at the expense that errors happening during log initialization wouldn't be reported. I believe this is a fair tradeoff since this prevents starting the Relay server, so this is caught in other places, too.